src/map/packets/char_update.cpp

```diff
@@ -87,7 +87,7 @@ CCharUpdatePacket::CCharUpdatePacket(CCharEntity* PChar)
     if (PChar->StatusEffectContainer->HasStatusEffect(EFFECT_TERROR))
         flag |= 0x08;
     if (PChar->allegiance > 1)
-        flag |= (PChar->allegiance << 5);
+        flag = (PChar->allegiance << 5);
```

Comment:
and actually, the default allegiance is 1, which is 0x20 when left shifted 5 times.  So this whole thing can be simplified down to uint8 flag = PChar-> allegiance << 5 and the terror check after